pkg/apply/taskrunner/runner.go

```diff
@@ -147,8 +167,10 @@ func (tsr *TaskStatusRunner) Run(
 
 			// send a status update to the running task, but only if the status
 			// has changed and the task is tracking the object.
-			if currentTask.Identifiers().Contains(id) {
-				currentTask.StatusUpdate(taskContext, id)
+			if currentTask != nil {
```

Comment:
currentTask can be nil here when we are still waiting for the StatusWatcher to become synchronized right?

Reply:
Yeah, if we get a status update before full synchronization, which is likely when there are multiple resources being watched.